modules/products/static-site/storage.tf

```diff
@@ -52,6 +52,19 @@ data "aws_iam_policy_document" "static_site_iam_storage_policy_document" {
       values   = [aws_cloudfront_distribution.static_site_distribution.arn]
     }
   }
+  statement {
+    actions = [
+      "s3:ListBucket"
+    ]
+    resources = [
+      "arn:aws:s3:::${aws_s3_bucket.static_site.id}/*"
```

Comment:
I don't think you need the /* on this resource as it's on the resource itself and not the objects in the bucket